puma/service/defaults/main.yml

```diff
@@ -41,3 +41,11 @@ puma_upstart_conf: >-
      if puma_user and puma_user != 'root'
      else '/etc/init/puma.conf'
   }}
+
+puma_upstart_system: "{{ ansible_distribution_release == 'trusty' or ansible_distribution_release == 'precise' }}"
+puma_systemd_system: "{{ not (puma_upstart_system|bool) }}"
```

Comment:
I would rename the to variables to
`puma_service_upstart` and `puma_service_systemd`

Maybe even would be to have a `puma_service_type` which may contain `system` or `upstart`, where the default depends on your heuristic.